frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx

```diff
@@ -12,87 +13,170 @@ import {
+        disable: state.poolStatus === POOL_PROGRESS.PROGRESS,
+        onClick: handleFinishButton,
+      },
+    ],
+    [POOL_PROGRESS.TIMEOUT]: [
+      {
+        id: 'modal-finish-action',
+        label: t('ceph-storage-plugin~Finish'),
+        type: ButtonType.submit,
+        variant: ButtonVariant.primary,
+        disable: state.poolStatus === POOL_PROGRESS.PROGRESS,
+        onClick: handleFinishButton,
+      },
+    ],
+    default: [
+      {
+        id: 'modal-cancel-action',
+        label: t('ceph-storage-plugin~Cancel'),
+        type: ButtonType.button,
+        variant: ButtonVariant.secondary,
+        onClick: cancel,
+      },
+      {
+        id: 'modal-confirm-action',
+        label: t(`ceph-storage-plugin~${footerPrimaryAction}`),
+        type: ButtonType.submit,
+        variant:
+          footerPrimaryAction === FooterPrimaryActions.DELETE
+            ? ButtonVariant.danger
+            : ButtonVariant.primary,
+        disable:
+          footerPrimaryAction !== FooterPrimaryActions.DELETE &&
+          checkRequiredValues(
+            state.poolName,
+            state.replicaSize,
+            state.volumeType,
+            isPoolManagementSupported,
+          ),
+        onClick: onSubmit,
+      },
+    ],
+  };
+
+  return (
     <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
-      <Button type="button" variant="secondary" data-test-id="modal-cancel-action" onClick={cancel}>
-        {t('ceph-storage-plugin~Cancel')}
-      </Button>
-      <Button
-        type="button"
-        variant="primary"
-        data-test="modal-create-action"
-        onClick={onClick}
-        isDisabled={checkRequiredValues(
-          state.poolName,
-          state.replicaSize,
-          state.volumeType,
-          isPoolManagementSupported,
-        )}
-      >
-        {actionLabel}
-      </Button>
+      {footerButtonsFactory[state.poolStatus || 'default'].map((buttonProp) => {
+        return (
+          <Button
+            type={buttonProp.type}
+            variant={buttonProp.variant}
+            isDisabled={buttonProp.disable}
+            id={buttonProp.id}
+            data-test={buttonProp.id}
+            onClick={buttonProp.onClick}
+          >
+            {buttonProp.label}
+          </Button>
+        );
+      })}
     </ActionGroup>
   );
 };
 
 type BlockPoolModalFooterProps = {
   state: BlockPoolState;
   dispatch: React.Dispatch<BlockPoolAction>;
-  onClick: () => void;
-  actionLabel: string;
-  closeLabel: string;
-  onPoolCreation?: (name: string) => void;
+  footerPrimaryAction: FooterPrimaryActions;
+  onSubmit: () => void;
 } & ModalComponentProps;
+
+type ButtonProps = {
+  id: string;
+  label: string;
+  type: ButtonType;
+  variant: ButtonVariant;
+  disable?: boolean;
+  onClick: (e?: React.FormEvent<EventTarget>) => void;
+};
+
+type ModalFooterButtonFactory = {
+  [stauts: string]: ButtonProps[];
```

Comment:
```suggestion
  [status: POOL_PROGRESS]: ButtonProps[];
```

Reply:
ack

Reply:
i think it should be status in  POOL_PROGRESS

Reply:
reason i used string is, i used default string